.github/workflows/hlint.yml

```diff
@@ -37,3 +37,32 @@ jobs:
           then
             echo "$CHANGED_HS_FILES" | xargs ${{env.working-directory}}/hlint --json --hint=${{env.working-directory}}/.hlint.yaml | jq -r "$JQ_SCRIPT"
           fi
+
+  import-checks:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Check imports
+        shell: bash
+        run: |
+          FOUND_ERRORS=0
+          function check_imports() {
+              rootfolder="$1"
+              forbidden="$2"
+              message="$3"
+              for match in $(find "server/src-lib/$rootfolder" -type f -iname "*.hs" | xargs grep -Hn -E "^import.*$forbidden" | cut -d ':' -f 1,2); do
```

Comment:
To be on the safe side, it would be great if this would only test files that are touched by the PR.